backend/controllers/user.controller.ts

```diff
@@ -0,0 +1,315 @@
+import dotenv from 'dotenv';
+import { Request, Response } from 'express';
+import { PrismaClient } from '@prisma/client';
+import otpGenerator from 'otp-generator';
+import jwt from 'jsonwebtoken';
+import nodemailer from 'nodemailer';
+
+const prisma = new PrismaClient();
+const result = dotenv.config();
+
+// could use refactoring later
+// nodemailer config -> transport
+const config = {
+    service: 'gmail',
+    auth: {
+        user: 'nomansanjari2001@gmail.com',
+        pass: 'ttfgnbtjykxedjzp'
+    }
+}
+
+// mail transporter
+let transporter = nodemailer.createTransport(config);
+
+// test User
+export const studentTest = async (req: Request, res: Response) => {
+    res.status(200).json({ success: true, message: 'Endpoint works' });
+}
+
+// create new User
+export const createNewStudent = async (req: Request, res: Response) => {
+    const { schoolName,
+        email,
+        firstName,
+        lastName,
+        username,
+        yearOfStudy,
+        password } = req.body;
+
+    const domain = email.slice(email.indexOf('@') + 1);
+
+    // generating the OTP
+    const otp = otpGenerator.generate(6, {
+        upperCaseAlphabets: false,
+        lowerCaseAlphabets: false,
+        specialChars: false,
+    });
+
+    // verify if school is accepted
+    const schoolID = await prisma.school.findFirst({
+        where: {
+            name: schoolName,
+            domain: domain
+        },
+    });
+
+    // verify if email doesnt exist
+    const studentExists = await prisma.user.findFirst({
```

Comment:
this also needs to make sure the student status is true. If it already exists but the status is False, it should just follow the regular flow of sending the OTP but it shouldn't create a new student object just use the pre-existing one